pubpub/pubpub
# feat: initial reference work; link to internal pub nodes w/ ids

This PR accomplishes two main things: 
- duplicates the baseline functionality of references (still WIP) from the `reactive-demo` branch
- adds the ability to link to any node with an id in the editor, including heading, paragraph, audio, video, image, file, citation, footnote, equation, and table nodes.

![captured](https://user-images.githubusercontent.com/6402908/92526088-e8e62d00-f1f2-11ea-91d1-26abf4ba9e4e.gif)

Any node with an id at the top level of the document is eligible for linking. Linking works identically to how it did prior to this PR. The PubMouseEvents component now uses event delegation to avoid potential performance issues in documents with a large number of nodes with ids (which now include paragraphs, too!). To get event delegation working, I switched from using "mouseenter" and "mouseleave" events to the bubbling "mouseover" and "mouseout" events.

This PR also adds typing of props of the Menu/MenuItem components, and removes a bunch of `// @ts-expect-error` comments.

Test Plan:
- Create a document with one of each of the above node types (heading, paragraph, audio, video, image, file, citation, footnote, equation, and table).
- Ensure when the elements are hovered, a link button appears outside of the pub body column to the left of the element.
- Ensure that content-less paragraphs do not display the link button when hovered.
- Clicking on the link button should copy a "permanent" link to the element in the document.
- Move an element (i.e. copy + paste it) and ensure the link maintains the reference to the correct element.
- Export the Pub and ensure figures are rendered correctly.
